src/ripple/app/misc/AmendmentTableImpl.cpp

```diff
@@ -151,28 +236,27 @@ AmendmentTableImpl::get (std::string const& name)
     return uint256 ();
 }
 
-AmendmentState*
-AmendmentTableImpl::addKnown (const char* amendmentID, const char* friendlyName,
-    bool veto)
+bool
+AmendmentTableImpl::addKnown (uint256 const& hash,
+                              std::string const& friendlyName,
+                              bool veto)
 {
-    uint256 hash;
-    hash.SetHex (amendmentID);
-
     if (hash.isZero ())
     {
         assert (false);
-        return nullptr;
+        return false;
```

Comment:
As I mentioned elsewhere, it's not clear what the return value means.

Indeed, the semantics of an `assert` should mean that this path is _never_ taken because our program is in a bad state if it is.  So why bother returning anything?